src/components/tx/AdvancedParamsForm/AdvancedParams.tsx

```diff
@@ -0,0 +1,60 @@
+import GasParams from '@/components/tx/GasParams'
+import { useState } from 'react'
+import AdvancedParamsForm, { AdvancedParameters } from '@/components/tx/AdvancedParamsForm/index'
+
+type Props = AdvancedParameters & {
+  recommendedNonce?: number
+  willExecute: boolean
+  nonceReadonly: boolean
+  isEstimating: boolean
+  onFormSubmit: (data: AdvancedParameters) => void
+}
+
+const AdvancedParams = ({
+  nonce,
+  gasLimit,
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+  safeTxGas,
+  recommendedNonce,
+  willExecute,
+  nonceReadonly,
+  isEstimating,
+  onFormSubmit,
+}: Props) => {
+  const [isEditingGas, setIsEditingGas] = useState<boolean>(false)
+
+  const onAdvancedSubmit = (data: AdvancedParameters) => {
+    onFormSubmit(data)
+
+    setIsEditingGas(false)
+  }
+
+  return isEditingGas ? (
+    <AdvancedParamsForm
+      nonce={nonce}
+      gasLimit={gasLimit}
+      maxFeePerGas={maxFeePerGas}
+      maxPriorityFeePerGas={maxPriorityFeePerGas}
+      safeTxGas={safeTxGas}
+      isExecution={willExecute}
+      recommendedNonce={recommendedNonce}
+      estimatedGasLimit={gasLimit?.toString()}
+      nonceReadonly={nonceReadonly}
+      onSubmit={onAdvancedSubmit}
+    />
+  ) : (
+    <GasParams
+      isExecution={willExecute}
+      isLoading={isEstimating}
```

Comment:
`isEstimating` is now basically needed only for this, so I suggest we don't pass it at all. So no need to pass `gasPriceLoading` either. GasParams could just check if gasLimit and price are undefined to the same effect.

The only good thing it adds is also showing a skeleton when the gas price is re-fetched when polling.

Reply:
Good point, I removed the prop and instead check the values directly. Apart from the polling skeleton it looks fine to me. We will still need `gasPriceLoading` outside of it though to decide whether the Submit button should be disabled or not.

Reply:
Lets remove it entirely and only depend on `gasLimitLoading`. That way its only part of the advanced params.